### PR TITLE
mgmt: mcumgr: grp: os_mgmt: Add optional bootloader info hook

### PR DESCRIPTION
**Pull request by nordicjm**
Adds an optional hook that allows for appending additional
    responses to the bootloader info command

### CODE REVIEW

**Comment by nordicjm** [comment]
@tomchy

**Review by tomchy — APPROVED** [review]
Verified in combination with https://github.com/zephyrproject-rtos/zephyr/pull/78312 using https://github.com/nrfconnect/sdk-nrf/pull/17301.

**Review by de-nordic — APPROVED** [review]

**Comment by mmahadevan108** [comment]
Wanting some more reviews for this PR.
cc @danieldegrasse @pdgendt  @cfriedt @nashif  to see if they can take a look as well

File: subsys/mgmt/mcumgr/grp/os_mgmt/src/os_mgmt.c [hunk 1]
```diff
@@ -450,6 +450,17 @@ os_mgmt_bootloader_info(struct smp_streamer *ctxt)
...
 
+#if defined(CONFIG_MCUMGR_GRP_OS_BOOTLOADER_INFO_HOOK)
+	enum mgmt_cb_return status;
+	int32_t err_rc;
+	uint16_t err_group;
+	struct os_mgmt_bootloader_info_data bootloader_info_data = {
+		.zse = zse,
+		.decoded = &decoded,
+		.query = &query
+	};
```
Comment (pdgendt) [inline subsys/mgmt/mcumgr/grp/os_mgmt/src/os_mgmt.c:461]: Nit
```suggestion
	struct os_mgmt_bootloader_info_data bootloader_info_data = {
		.zse = zse,
		.decoded = &decoded,
		.query = &query,
	};
```

**Review by pdgendt — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**